### PR TITLE
[SPARK-14213][SQL] Migrate HiveQl parsing to ANTLR4 parser

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?

This PR migrates all HiveQl parsing to the new ANTLR4 parser. This PR is build on top of https://github.com/apache/spark/pull/12011, and we should wait with merging until that one is in (hence the WIP tag).

As soon as this PR is merged we can start removing much of the old parser infrastructure.
### How was this patch tested?

Exisiting Hive unit tests.

cc @rxin @andrewor14 @yhuai

### CODE REVIEW

**Comment by rxin** [comment]
Thanks - I've merged this in master.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    val sql = Option(source(query))
+    val tableDesc = CatalogTable(
+      identifier = visitTableIdentifier(name),
+      tableType = CatalogTableType.VIRTUAL_VIEW,
+      schema = schema,
+      storage = EmptyStorageFormat,
+      properties = properties,
+      viewOriginalText = sql,
+      viewText = sql)
+    CreateView(tableDesc, plan(query), allowExist, replace, command(ctx))
```
Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala]: @hvanhovell I search the whole code base, but still unable to find the implementation of this class `CreateView`? I am just wondering how Parser can recognize it? Thanks!

Comment (hvanhovell) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala]: @gatorsmile I renamed the import to `CreateView`. The actual class is `org.apache.spark.sql.hive.CreateViewAsSelect`. See the following line in `HiveSqlParser`:

    import org.apache.spark.sql.hive.{CreateTableAsSelect => CTAS, CreateViewAsSelect => CreateView}

Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala]: NVM, `CreateViewAsSelect => CreateView`. I found it. Thanks!

**Pull request closed without merging**